cmd/kube-proxy/app/server_others.go

```diff
@@ -108,14 +108,7 @@ func newProxyServer(
 	iptInterface = utiliptables.New(execer, protocol)
 	kernelHandler = ipvs.NewLinuxKernelHandler()
 	ipsetInterface = utilipset.New(execer)
-	canUseIPVS, err := ipvs.CanUseIPVSProxier(kernelHandler, ipsetInterface)
-	if string(config.Mode) == proxyModeIPVS && err != nil {
-		klog.Errorf("Can't use the IPVS proxier: %v", err)
-	}
-
-	if canUseIPVS {
-		ipvsInterface = utilipvs.New(execer)
-	}
+	ipvsInterface = utilipvs.New(execer)
```

Comment:
Always setting the ipvs interface before checking if we can use it would break kube-proxy --cleanup.

Reply:
> Always setting the ipvs interface before checking if we can use it would break kube-proxy --cleanup.

Sorry, do you mean we should only do `ipvs.CleanupLeftovers` when we choose ipvs mode? It seems not consistent with other modes.

Reply:
Doesn't --cleanup do every leftover cleanup?

Reply:
> Sorry, do you mean we should only do ipvs.CleanupLeftovers when we choose ipvs mode? It seems not consistent with other modes.

No I mean that `--cleanup` runs `CleanupLeftovers` for all the proxy modes and it does a `nil` check on the ipvs interface we pass to it. If we always set it, `--cleanup` could error because we would try to flush ipvs rules where IPVS is not available

Reply:
Get it, I'll update it soon.